### PR TITLE
Major: Rewrite for Companion 3.0

### PR DESCRIPTION
**Pull request by bryce-seifert**

### CODE REVIEW

File: actions.js [hunk 1]
```diff
@@ -1,876 +1,947 @@
...
+			options: [
+				{
+					type: 'textinput',
+					useVariables: true,
+					label: 'Cue Name',
+					id: 'string',
+				},
+			],
+			callback: (action) => {
+				this.parseVariablesInString(action.options.string).then((value) => {
```
Comment (Julusian) [inline actions.js:197]: You could write this as:
```
callback: async (action) => {
	const value = await this.parseVariablesInString(action.options.string)
	this.sendCommand('jumpToCueWithName', value)
},
```

File: presets.js [hunk 2]
```diff
@@ -18,1099 +20,1444 @@ exports.getPresets = function () {
...
 				color: ColorWhite,
 				bgcolor: ColorBlack,
 			},
-			actions: [
+			steps: [
 				{
-					action: 'toggle_play',
+					down: [
+						{
+							actionId: 'toggle_play',
```
Comment (Julusian) [inline presets.js:151]: Do these presets work? I have a feeling they will error without an options property, even if it is empty

Comment (bryce-seifert) [reply presets.js:151]: They do work, but I'll go ahead and add an empty options property to be safe / future proof

File: actions.js (hunk 1 above)
Comment (bryce-seifert) [reply actions.js:197]: @Julusian Addressed this with https://github.com/bitfocus/companion-module-imimot-mitti/pull/25

File: presets.js (hunk 2 above)
Comment (bryce-seifert) [reply presets.js:151]: Addressed with https://github.com/bitfocus/companion-module-imimot-mitti/pull/25

**Pull request merged**